.travis.yml

```diff
@@ -19,4 +21,15 @@ matrix:
       env: TEST_SUITE=lint
     - node_js: "4"
       env: TEST_SUITE=coveralls
-script: npm run $TEST_SUITE
+    - node_js: "4"
+      env: TEST_SUITE=saucelabs
+  allow_failures:
+    - env: TEST_SUITE=saucelabs
+fast_finish: true
+cache:
+  directories:
+    - node_modules
+before_script:
+  - npm install -g grunt-cli
+script:
+  - ./travis.sh
```

Comment:
As I understand you don't need `travis.sh`, put instead `TEST_SUITE` full command (`npm run unit`)